pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/ZkBasedTableRebalanceObserver.java

```diff
@@ -126,30 +143,74 @@ public int getNumUpdatesToZk() {
     return _numUpdatesToZk;
   }
 
+  @VisibleForTesting
+  TableRebalanceRetryConfig getTableRebalanceJobRetryConfig() {
+    return _tableRebalanceJobRetryConfig;
+  }
+
   private void trackStatsInZk() {
+    Map<String, String> jobMetadata =
+        createJobMetadata(_tableNameWithType, _rebalanceJobId, _tableRebalanceProgressStats,
+            _tableRebalanceJobRetryConfig);
+    _pinotHelixResourceManager.addControllerJobToZK(_rebalanceJobId, jobMetadata,
+        ZKMetadataProvider.constructPropertyStorePathForControllerJob(ControllerJobType.TABLE_REBALANCE),
+        prevJobMetadata -> {
+          // Abort the job when we're sure it has failed, otherwise continue to update the status.
+          if (prevJobMetadata == null) {
+            return true;
+          }
+          String prevStatsInStr = prevJobMetadata.get(RebalanceJobConstants.JOB_METADATA_KEY_REBALANCE_PROGRESS_STATS);
+          TableRebalanceProgressStats prevStats;
+          try {
+            prevStats = JsonUtils.stringToObject(prevStatsInStr, TableRebalanceProgressStats.class);
+          } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+          }
```

Comment:
should we log the exception and return false?

Reply:
oh, good catch, this should be `return true`, to let current rebalance continue. As commented above, we abort if we really get a FAILED status, otherwise, just be conservative and let it continue.